Configurations/BaseCommon.xcconfig

```diff
@@ -0,0 +1,41 @@
+// Copyright (C) 2022 Apple Inc. All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions
+// are met:
+// 1. Redistributions of source code must retain the above copyright
+//    notice, this list of conditions and the following disclaimer.
+// 2. Redistributions in binary form must reproduce the above copyright
+//    notice, this list of conditions and the following disclaimer in the
+//    documentation and/or other materials provided with the distribution.
+//
+// THIS SOFTWARE IS PROVIDED BY APPLE INC. ``AS IS'' AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL APPLE INC. OR
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+// OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+// This configuration file should be used to set defaults for all WebKit projects.
+
+#include "Sanitizers.xcconfig"
+// SDKVariant.xcconfig is included via Sanitizers.xcconfig.
+
+GCC_OPTIMIZATION_LEVEL = $(GCC_OPTIMIZATION_LEVEL_WK_OVERRIDE_$(WK_OVERRIDE_GCC_OPTIMIZATION_LEVEL));
+GCC_OPTIMIZATION_LEVEL_WK_OVERRIDE_YES = $(WK_GCC_OPTIMIZATION_LEVEL);
+// Define GCC_OPTIMIZATION_LEVEL_WK_OVERRIDE_NO in each project's Base.xcconfig.
```

Comment:
I think it would be more understandable if this were named `WK_DEFAULT_GCC_OPTIMIZATION_LEVEL` or simply `WK_GCC_OPTIMIZATION_LEVEL`, though the latter would mean shuffling arround the name of the setting used below.

Reply:
Yeah, after looking at this patch again, I think I prefer `WK_GCC_OPTIMIZATION_LEVEL` as well.  Thanks!

Reply:
Actually, I think I'll use `WK_BASE_GCC_OPTIMIZATION_LEVEL` to match how `WK_BASE_` will be used for `OTHER_*` variables.